src/Switch.tsx

```diff
@@ -0,0 +1,358 @@
+  sx: sxProp
+}) => {
+  const isControlled = typeof defaultOn === 'undefined' && typeof onProp !== 'undefined'
+  const {theme, colorScheme} = useTheme()
+  const [onState, setOnState] = useProvidedStateOrCreate<boolean>(onProp, onChange, Boolean(defaultOn))
+  const acceptsInteraction = !disabled && !isLoading
+  const handleToggleClick: MouseEventHandler = useCallback(
+    e => {
+      if (!isControlled) {
+        setOnState(!onState)
+      }
+      onClick && onClick(e)
+    },
+    [onClick, isControlled, onState, setOnState]
+  )
+
+  useEffect(() => {
+    if (onChange && isControlled) {
+      onChange(Boolean(onProp))
+    }
+  }, [onChange, onProp, isControlled])
+
+  return (
+    <Box
+      display="inline-flex"
+      alignItems="center"
+      flexDirection={statusLabelPosition === 'start' ? 'row' : 'row-reverse'}
+      mb={4}
+      sx={sxProp}
+    >
+      {isLoading ? <Spinner size="small" /> : null}
+      <Text
+        color={acceptsInteraction ? 'fg.default' : 'fg.muted'}
+        fontSize={size === 'small' ? 0 : 1}
+        mx={2}
+        aria-hidden="true"
+        sx={{position: 'relative'}}
+      >
+        <Box textAlign="right" sx={onState ? null : hiddenTextStyles}>
+          On
+        </Box>
+        <Box textAlign="right" sx={onState ? hiddenTextStyles : null}>
+          Off
+        </Box>
+      </Text>
+      <SwitchButton
+        onClick={handleToggleClick}
+        aria-labelledby={ariaLabelledby}
+        aria-describedby={ariaDescribedby}
+        aria-checked={onState}
+        aria-disabled={!acceptsInteraction}
+        role="switch"
+        on={onState}
+        size={size}
+        disabled={!acceptsInteraction}
+        isHighContrast={Boolean(colorScheme?.includes('high_contrast'))}
+      >
+        <VisuallyHidden>{onState ? 'On' : 'Off'}</VisuallyHidden>
+        <Box aria-hidden="true" display="flex" alignItems="center" width="100%" height="100%" overflow="hidden">
+          <Box
+            flexGrow={1}
+            flexShrink={0}
+            flexBasis="50%"
+            color={acceptsInteraction ? 'accent.fg' : 'fg.subtle'}
+            lineHeight="0"
+            sx={{
+              transform: `translateX(${onState ? '0' : '-100%'})`,
+              transitionProperty: 'transform',
+              transitionDuration: TRANSITION_DURATION,
+              '> svg': {
+                fill: 'currentcolor'
+              }
+            }}
+          >
+            <LineIcon size={size} />
+          </Box>
+          <Box
+            flexGrow={1}
+            flexShrink={0}
+            flexBasis="50%"
+            color={acceptsInteraction ? 'fg.default' : 'fg.subtle'}
+            lineHeight="0"
+            sx={{
+              transform: `translateX(${onState ? '100%' : '0'})`,
+              transitionProperty: 'transform',
+              transitionDuration: TRANSITION_DURATION,
+              '> svg': {
+                fill: 'currentcolor'
+              }
```

Comment:
Is this necessary? Looks like the SVG already has a fill of `currentColor`

```suggestion
```

Reply:
You're right. I forgot to remove this when I made that change.